src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -0,0 +1,304 @@
+            if (!$hasEmployeeEditPermission) {
+                $bulkConfiguration[$profile['id']] = [
+                    'view' => false,
+                    'add' => false,
+                    'edit' => false,
+                    'delete' => false,
+                    'all' => false,
+                ];
+
+                continue;
+            }
+
+            foreach ($tabs as $tab) {
+                foreach ($permissions as $permission) {
+                    if (!$profileTabPermissions[$employeeProfileId][$tab['id']][$permission]) {
+                        $bulkConfiguration[$profile['id']]['view'] = false;
+                        $bulkConfiguration[$profile['id']]['all'] = false;
+
+                        break;
+                    }
+                }
+
+                foreach ($tab['children'] as $childTab) {
+                    foreach ($permissions as $permission) {
+                        if (!$profileTabPermissions[$employeeProfileId][$childTab['id']][$permission]) {
+                            $bulkConfiguration[$profile['id']]['add'] = false;
+                            $bulkConfiguration[$profile['id']]['all'] = false;
+
+                            break;
+                        }
+                    }
+
+                    foreach ($childTab['children'] as $subChild) {
+                        foreach ($permissions as $permission) {
+                            if (!$profileTabPermissions[$employeeProfileId][$subChild['id']][$permission]) {
+                                $bulkConfiguration[$profile['id']]['edit'] = false;
+                                $bulkConfiguration[$profile['id']]['all'] = false;
+
+                                break;
+                            }
+                        }
+
+                        foreach ($subChild['children'] as $subSubChild) {
+                            foreach ($permissions as $permission) {
+                                if (!$profileTabPermissions[$employeeProfileId][$subSubChild['id']][$permission]) {
+                                    $bulkConfiguration[$profile['id']]['delete'] = false;
+                                    $bulkConfiguration[$profile['id']]['all'] = false;
+
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return $bulkConfiguration;
+    }
+
+    /**
+     * @param array $profiles
+     *
+     * @return array
+     */
+    private function getModulePermissionsForProfiles(array $profiles)
+    {
+        $profilePermissionsForModules = [];
+
+        foreach ($profiles as $profile) {
+            $profilePermissionsForModules[$profile['id']] = Module::getModulesAccessesByIdProfile($profile['id']);
+            uasort($profilePermissionsForModules[$profile['id']], function ($a, $b) {
+                return strnatcmp($a['name'], $b['name']);
```

Comment:
I had once a bug because of a corrupted module (something like he was uninstalled but still registered) so the key `name` was empty. So ... please check `$a['name']` exists ... It's sad but I know this is a actual usecase we have to handle

Reply:
Again, seems like we want to make our code bug tolerant, the sooner we let user know that something is messed up in PS the better, otherwise we fix something that we should not. :/

Reply:
Unfortunately, we need to make PS bug tolerant when it comes to modules because this is an area where user is quite free to mess things up.

Basically, if the user messed up one of this module, he should still be able to use the Permission page 😉. The best would be also a notice message but in a Handler it's going to be hard to do, isn't it ?

Reply:
For me it seems that it's not something that module can break. Why do we get module with empty name in the first place? Maybe it's because Module management component allows it? Or maybe install/uninstall process has a bug? Or maybe `Module::getModulesAccessesByIdProfile` has a bug instead? I think we should fix the cause of it and not the "client".

Reply:
We should fix both 😛this is [defensive programming](https://en.wikipedia.org/wiki/Defensive_programming) I want to apply: do not trust other parts of the software. Expect database failures, expect other parts of the code to be updated in a bad way, expect bad filesystem access settings. And handle these bad usecases in a decent way.

This is what we already do in Command Handlers, [example with AddCurrencyHandler](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Currency/CommandHandler/AddCurrencyHandler.php#L61).

If the failure is linked to current processing, the good answer is to fail gracefully, with a nice error message when relevant. However here the data is corrupted. Since we are trying to sort this corrupted data, not process it, I think the safe way is to ignore it or provide a default/fallback behavior. If I screw up an Employee data I still want to be able to list my other, fine Employees. The corrupted Employee should either be ignored or displayed with a warning.

Reply:
Ok @matks, I'll update this, however, I don't really agree with

> do not trust other parts of the software

Here's why. First of all, I agree with defensive programming, but it must be reasonable. Here's an example:

```php
class SomeClass {
    public function doSomething($cart) {
        // let's assume we can't use typehints
        // so we are making sure it's cart before going on
        // which I think is good example of defensive programming
        assertInstanceOf(Cart::class, $cart);

        // ...
    }
}
```
> Expect database failures, expect other parts of the code to be updated in a bad way, expect bad filesystem access settings. And handle these bad usecases in a decent way.

I think we are crossing boundary between "Defensive programing" and "Paranoid Programing" here. We can't add assertions, checks & various validations everywhere, we would end up with more validations than business logic. From you suggestion, I understand that we should do something like:

```php
function doFoo(array $data) {
    // we don't know where $data comes from
    if (!isset($data['name']) {
        // fix or throw exception
    }

    // ...

    doFar($data);
}

function doFar(array $data) {
    // we still don't know where $data comes from
    if (!isset($data['name']) {
        // fix or throw exception
    }

    // ...

    // would contain yet another validation for $data
    // because it doesn't know where it comes from
    doBar($data);
}
```

It reminds me of [Defensive Programming vs. Batshit Crazy Paranoid Programming ](https://dzone.com/articles/defensive-programming-just), wdyt?

Reply:
In your example, if `doFoo` and `doFar` are in 2 classes you wrote, covered by unit and integration tests I agree it's paranoid.

But in our case, `doFar` is migrated code while `doFoo` is legacy code dealing with a complex part of prestashop (modules and how they integrate with the software) so I consider it defensive rather than paranoid 😅 .

However when we migrate this legacy code, I agree we can remove this code. But until then ... let's be safe 😄.

Reply:
Added checks as well, can you review? Also, should we document this decision in code as well? For someone else who is not familiar with this "issue" it might look weird.

Reply:
Agreed you can add a comment to explain this is a defensive check due to the modules that are not thrust worthy, also could you write them in single line?
`$a['name'] = !isset($a['name']) ? '' : $a['name'];`

Reply:
@sarjon only left is comment and shorter code here

Reply:
ok @jolelievre, ill update it tomorrow, but I think it's waste of time coming back these kind of "issues" to make legacy code prettier? We should focus on more important issues, especially now, before freeze. :/

Reply:
Even though it's an adapter we can't exactly call it legacy code any more, and I was just suggesting a 30 second modification along with the documentation you proposed to add
But you're right, let's focus on the important issues => PR approved

Reply:
@jolelievre I've addressed your comment, however, I still don't see value in it (in this particular case). Making `if` inline may look cooler at first, but it does not contribute to code readability as cognitive load is much higher for developer who's reading this code rather than reading simple `if` block imo.